tests/by-util/test_pgrep.rs

```diff
@@ -0,0 +1,83 @@
+// This file is part of the uutils procps package.
+//
+// For the full copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+use crate::common::util::TestScenario;
+
+#[test]
+fn test_invalid_arg() {
+    new_ucmd!().arg("--definitely-invalid").fails().code_is(1);
+}
+
+#[test]
+fn test_pgrep() {
+    new_ucmd!().arg("--help").succeeds().code_is(0);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_oldest() {
+    new_ucmd!().arg("-o").succeeds().code_is(0);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_newest() {
+    new_ucmd!().arg("-n").succeeds().code_is(0);
+}
+
+#[test]
+fn test_not_exist_program() {
+    new_ucmd!()
+        .arg("THIS_PROGRAM_DOES_NOT_EXIST")
+        .fails()
+        .code_is(1);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_full() {
+    new_ucmd!().arg("sh").arg("--full").succeeds().code_is(0);
+}
+
+#[test]
+#[cfg(target_os = "linux")]
+fn test_invalid_regex() {
+    new_ucmd!().arg("{(*").arg("--exact").fails().code_is(2);
```

Comment:
While this test works fine, if you omit the `--exact` argument it fails. The behavior should be the same in both cases.

Reply:
Hmmm ...... It looks like I didn't quite understand what you meant, could you please be more detailed?

Reply:
Sure, what I wanted to say is that the output and exit code should be the same for `pgrep {(* --exact` and `pgrep {(*`. Currently, they are different.

Reply:
I'll add it to the test later。

Reply:
So I think it should be

```rust
#[test]
#[cfg(target_os = "linux")]
fn test_invalid_regex() {
    new_ucmd!().arg("{(*").arg("--exact").fails().code_is(2);
    new_ucmd!().arg("{(*").fails().code_is(2);
}
```

does it right?

Reply:
That looks good :)